httpcli/basic/basic.go

```diff
@@ -9,16 +9,26 @@ import (
 
 // Register registers a DoerFactory for HTTP Basic authentication
 func Register() {
-	httpcli.Register(httpcli.AuthBasic, httpcli.DoerFactory(func(cm httpcli.ConfigMap, c *http.Client) httpcli.Doer {
+	httpcli.Register(httpcli.AuthBasic, httpcli.DoerFactory(func(cm httpcli.ConfigMap, c *http.Client) (doer httpcli.Doer) {
 		obj := cm.FindOrPanic(httpcli.AuthBasic)
 		config, ok := obj.(Credentials)
 		if !ok {
-			panic(fmt.Sprintf("expected Credentials instead of %#+v", obj))
+			panic(fmt.Errorf("expected Credentials instead of %#+v", obj))
 		}
-		return Doer(c, config)
+		validate(config)
+		if c != nil {
```

Comment:
if c is nil then we should probably panic. Or, we should modify this function to return an error.